hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/managed/ManagedSSTDumpIterator.java

```diff
@@ -0,0 +1,203 @@
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Iterator to Parse output of RocksDBSSTDumpTool.
+ */
+public class ManagedSSTDumpIterator implements
+        Iterator<ManagedSSTDumpIterator.KeyValue>, AutoCloseable {
+  private Process process;
+  private static final String SST_DUMP_TOOL_CLASS =
+          "org.apache.hadoop.hdds.utils.db.managed.ManagedSSTDumpTool";
+  private static final String PATTERN_REGEX =
+          "'([^=>]+)' seq:([0-9]+), type:([0-9]+) =>";
+
+  public static final int PATTERN_KEY_GROUP_NUMBER = 1;
+  public static final int PATTERN_SEQ_GROUP_NUMBER = 2;
+  public static final int PATTERN_TYPE_GROUP_NUMBER = 3;
+  private static final Pattern PATTERN_MATCHER =
+          Pattern.compile(PATTERN_REGEX);
+  private BufferedReader processOutput;
+  private StringBuilder stdoutString;
+
+  private Matcher currentMatcher;
+  private int prevMatchEndIndex;
+  private KeyValue currentKey;
+  private char[] charBuffer;
+  private KeyValue nextKey;
+
+  private long pollIntervalMillis;
+  private Lock lock;
+
+
+  public ManagedSSTDumpIterator(String sstDumptoolJarPath,
+                                String sstFilePath,
+                                long pollIntervalMillis) throws IOException {
+    File sstFile = new File(sstFilePath);
+    if (!sstFile.exists() || !sstFile.isFile()) {
+      throw new IOException(String.format("Invalid SST File Path : %s",
+              sstFile.getAbsolutePath()));
+    }
+    this.pollIntervalMillis = pollIntervalMillis;
+    this.lock = new ReentrantLock();
+    init(sstFile, sstDumptoolJarPath);
+  }
+
+  private void init(File sstFile, String sstDumptoolJarPath)
+          throws IOException {
+    List<String> args = Lists.newArrayList(
+            "--file=" + sstFile.getAbsolutePath(),
+            "--command=scan");
+    process = HddsServerUtil.getJavaProcess(Collections.emptyList(),
+            sstDumptoolJarPath, SST_DUMP_TOOL_CLASS, args).start();
+    processOutput = new BufferedReader(new InputStreamReader(
+            process.getInputStream(), StandardCharsets.UTF_8));
+    stdoutString = new StringBuilder();
+    currentMatcher = PATTERN_MATCHER.matcher(stdoutString);
+    charBuffer = new char[8192];
+    next();
+  }
+
+  private void checkSanityOfProcess() {
+    if (!process.isAlive() && process.exitValue() != 0) {
+      throw new RuntimeException("Process Terminated with non zero " +
+              String.format("exit value %d", process.exitValue()));
+    }
+  }
+
+  @Override
+  public boolean hasNext() {
+    checkSanityOfProcess();
+    return nextKey != null;
+  }
+
+  @Override
+  public KeyValue next() throws RuntimeIOException {
```

Comment:
Unchcked exception like `RuntimeIOException` does not need to be declared.

```suggestion
  public KeyValue next() {
```